metals/src/main/scala/scala/meta/internal/metals/ForwardingMetalsBuildClient.scala

```diff
@@ -37,30 +50,38 @@ final class ForwardingMetalsBuildClient(
     clientConfig: ClientConfiguration,
     statusBar: StatusBar,
     time: Time,
-    didCompile: CompileReport => Unit,
-    onBuildTargetDidCompile: BuildTargetIdentifier => Unit,
+    didCompile: b.CompileReport => Unit,
+    onBuildTargetDidCompile: b.BuildTargetIdentifier => Unit,
     onBuildTargetDidChangeFunc: b.DidChangeBuildTarget => Unit,
 ) extends MetalsBuildClient
     with Cancelable {
 
+  private val forwarders =
+    new ju.concurrent.atomic.AtomicReference(ListBuffer.empty[LogForwarder])
```

Comment:
Not sure if this has improved the thread safety of this code, or if it is even safe at all. We are wrapping a mutable `ListBuffer` instance in an atomic reference, could it be a better idea to wrap an immutable collection instead? But maybe I misunderstood the intent. :slightly_smiling_face:

Reply:
indeed, that was the intend, should be `new ju.concurrent.atomic.AtomicReference(List.empty[LogForwarder])`